src/main/java/alfred/manager/TaskManager.java

```diff
@@ -0,0 +1,145 @@
+package alfred.manager;
+
+import alfred.exception.EmptyDescriptionException;
+import alfred.exception.InvalidDateException;
+import alfred.task.Deadline;
+import alfred.task.Event;
+import alfred.task.Task;
+import alfred.task.Todo;
+
+import java.util.ArrayList;
+
+public class TaskManager {
+    private ArrayList<Task> tasks;
+
+    public TaskManager() {
+        tasks = new ArrayList<>();
+        DataManager.populateTasks(tasks);
+        MessageManager.initMessage();
+    }
+
+    public void processInput(String userInput) {
+        String[] destructuredInputs = userInput.split(" ");
+        try {
+            switch (destructuredInputs[0]) {
+            case "list":
+                listTasks();
+                break;
+            case "done":
+                completeTask(userInput);
+                break;
+            case "todo":
+                // Fallthrough
```

Comment:
The cases for "event" and "todo" are falling through, the method does not do anything when these two commands show up. However, the methods for addTodo and addEvent are implemented and used in the addTask method. Is this fallthrough a mistake or the methods for Todo and Event are not ready yet?

Reply:
Hi Mick, I was concerned with the length of my `processInput()` function. Initially, I had implemented `addTodo()`, `addEvent()`, and `addDeadline()` within `processInput()`. However, due to exceptions handling, the length of my `processInput()` function exceeded 30 lines, and I decided to extract these three functions and handle them separately in `addTask()`. For this case, I specified a fallthrough so that the case of Todos, Events, and Deadlines, will be only handled by `addTask()`.

Reply:
I understand. Your rationale is valid to me.